docs/regression-testing-pipeline.md

```diff
@@ -0,0 +1,166 @@
+# F# Compiler Regression Testing Pipeline
+
+This document describes the F# compiler regression testing pipeline implemented in `azure-pipelines-regression-test.yml`.
+
+## Purpose
+
+The regression testing pipeline helps catch F# compiler regressions by building popular third-party F# libraries with the freshly built compiler from this repository. This provides early detection of breaking changes that might affect real-world F# projects.
+
+## How It Works
+
+### Pipeline Workflow
+
+1. **Build F# Compiler**: The pipeline first builds the F# compiler and packages from the current source code using the standard build process.
+
+2. **Test Against Third-Party Libraries**: For each library in the test matrix:
+   - Checkout the third-party repository at a specific commit
+   - Inject the `UseLocalCompiler.Directory.Build.props` configuration to use the locally built compiler
+   - Run the library's build script
+   - Capture detailed build logs and artifacts
+
+3. **Report Results**: Success/failure status is reported with detailed logs for diagnosis.
+
+### Key Features
+
+- **Reproducible Testing**: Uses specific commit SHAs for third-party libraries to ensure consistent results
+- **Matrix Configuration**: Supports testing multiple libraries with different build requirements
+- **Detailed Logging**: Captures comprehensive build logs, binary logs, and environment information
+- **Artifact Publishing**: Publishes build outputs for analysis when builds fail
+
+## Current Test Matrix
+
+The pipeline currently tests against:
+
+| Library | Repository | Commit | Build Script | Purpose |
+|---------|------------|--------|--------------|---------|
+| FSharpPlus | fsprojects/FSharpPlus | f614035b75922aba41ed6a36c2fc986a2171d2b8 | build.cmd | Tests advanced F# language features |
```

Comment:
[FsToolkit.ErrorHandling](https://github.com/demystifyfp/FsToolkit.ErrorHandling) might be another candidate.

Reply:
Once I get this working, I will do a callout and will invite people to add their projects with some limited/relaxed conditions.

Main criteria would be broadening coverage - so especially looking for projects that are using F# in ways that are different to this compiler and existing test suites.

Reply:
FS.EH and F#+ would be representatives for SRTPs. From the top of my head, those would be examples of "dimensions" I would be interested in covering with real world libraries/apps :

Quotations
Linq
ResumableCode
TypeProviders - authoring as well as consumption
Low-level code (ByRefs, Spans,..), incl. focus on related .NET/C# interop
Heavy CE-consumption